### PR TITLE
Refactored

### PR DESCRIPTION
**Pull request by mirii1994**

### CODE REVIEW

File: docker-guardian/README.md [hunk 1]
```diff
@@ -8,5 +8,15 @@ curl -sSL https://raw.githubusercontent.com/logzio/code-guardian/master/docker-g
 source ~/.zshrc
 ```
 
-That's it!
+### That's it!
 
+## Adding Customized Tests
```
Comment (yyyogev) [inline docker-guardian/README.md]: @mirii1994 maybe checks instead of tests?

File: docker-guardian/guardian-script.py [hunk 2]
```diff
@@ -20,67 +24,69 @@ def walk_in_directory():
...
+                search_regex(file_path, line, line_index)
                 line = fp.readline()
-                cnt = 1
-                while line:
-                    search_regex(file_path, line, cnt)
-                    line = fp.readline()
-                    cnt += 1
+                line_index += 1
     except:
         pass
```
Comment (yyyogev) [inline docker-guardian/guardian-script.py]: Add error message

File: docker-guardian/guardian-script.py [hunk 3]
```diff
@@ -20,67 +24,69 @@ def walk_in_directory():
...
 
 
-def search_regex(path, line, cnt):
+def search_regex(path, line, line_index):
```
Comment (yyyogev) [inline docker-guardian/guardian-script.py]: perhaps rename? check_sensitive? search_sensitive_data?

File: utils.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    return any(char.isupper() for char in exp)
+
+
+def contains_digit(exp):
+    return any(char.isdigit() for char in exp)
+
+
+def contains_special_char(exp):
+    invalid_chars = string.punctuation
+    return any(char in invalid_chars for char in exp)
```
Comment (yyyogev) [inline utils.py]: Nice one!

**Comment by yyyogev** [comment]
@mirii1994 please edit the quickstart files to add the utils.py file

File: docker-guardian/guardian-script.py (hunk 3 above)
Comment (mirii1994) [reply docker-guardian/guardian-script.py]: done

File: docker-guardian/guardian-script.py [hunk 5]
```diff
@@ -28,14 +28,15 @@ def check_file_data(file_path):
             line = fp.readline()
             line_index = 1
             while line:
-                search_regex(file_path, line, line_index)
+                check_sensitive(file_path, line, line_index)
                 line = fp.readline()
                 line_index += 1
     except:
+        logging.error("Could not read file: {}".format(file_path))
```
Comment (yyyogev) [inline docker-guardian/guardian-script.py]: Doesn't it makes the pass redundant?
And error -> warning

**Pull request merged**